sdk/go/common/util/gitutil/git_test.go

```diff
@@ -342,3 +381,95 @@ func TestTryGetVCSInfoFromSSHRemote(t *testing.T) {
 		assert.Equal(t, test.WantVCSInfo, got)
 	}
 }
+
+func TestParseAuthURL(t *testing.T) {
+	t.Parallel()
+
+	//nolint: gosec
+	generateSSHKey := func(t *testing.T, passphrase string) string {
+		// TODO(go1.21): clear(sshKeys)
+		for m := range sshKeys {
+			delete(sshKeys, m)
+		}
+
+		r := rand.New(rand.NewSource(0))
+		key, err := rsa.GenerateKey(r, 256)
+		require.NoError(t, err)
```

Comment:
Consider setting up the source of randomness outside the function so that two consecutive generateSSHKey calls have different-enough random input.
```suggestion

	r := rand.New(rand.NewSource(0))

	//nolint: gosec
	generateSSHKey := func(t *testing.T, passphrase string) string {
		// TODO(go1.21): clear(sshKeys)
		for m := range sshKeys {
			delete(sshKeys, m)
		}

		key, err := rsa.GenerateKey(r, 256)
		require.NoError(t, err)
```

Related optional nit: Consider shadowing `rand` with the result of `rand.New` if possible so that `rand` package functions are not used accidentally.

```go
	rand := rand.New(rand.NewSource(0))
```

Reply:
Just out of curiosity is it safe to share the source across parallel tests like this (For these tests the randomness doesn't really matter, we just need a small valid key.)

Reply:
Oh, good call. Yeah, rand.Source is not safe for concurrent use without a lock. What you were doing should be fine.